src/Common.cmake

```
@@ -69,8 +69,35 @@ macro(common_project_options)
         CACHE STRING "Fallbacks for the RelWithDebInfo build type")
   endif()
 
-  # Generate compile_commands.json to make it easier to work with clang based tools
-  set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
+  if(CMAKE_GENERATOR MATCHES ".*Makefile*." OR CMAKE_GENERATOR MATCHES ".*Ninja*")
+    # Generate compile_commands.json to make it easier to work with clang based tools
+    set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
+
+    # Make a symbol link of compile_commands.json on the source dir to help clang based tools find it
+    if(WIN32)
+      # Detect whether cmake is run as administrator (only administrator can read the LOCAL SERVICE account reg key)
+      cmake_host_system_information(RESULT IS_ADMINISTRATOR QUERY WINDOWS_REGISTRY "HKU\\S-1-5-19")
+
+      if(IS_ADMINISTRATOR)
+        # For administrator, symlink is available
+        file(CREATE_LINK ${CMAKE_BINARY_DIR}/compile_commands.json ${CMAKE_SOURCE_DIR}/compile_commands.json SYMBOLIC)
+      else()
+        # For non-administrator, create an auxiliary target and ask user to run it
+        add_custom_command(OUTPUT ${CMAKE_BINARY_DIR}/compile_commands.json
+          COMMAND ${CMAKE_COMMAND} -E copy ${CMAKE_BINARY_DIR}/compile_commands.json ${CMAKE_SOURCE_DIR}/compile_commands.json
+          DEPENDS ${CMAKE_SOURCE_DIR}/compile_commands.json
+          VERBATIM
+        )
+        add_custom_target(_copy_compile_commands
+          DEPENDS ${CMAKE_BINARY_DIR}/compile_commands.json
+          VERBATIM
+        )
+        message(AUTHOR_WARNING "Can't symlink compile_commands.json. Run cmake as administrator or use `cmake --build <build_dir> -t _copy_compile_commands` to copy it.")
```

Since this feature is optional, I would prefer a non-warning message.
```suggestion
        message(STATUS "compile_commands was not symlinked to the root. Run `cmake --build <build_dir> -t _copy_compile_commands` if needed.")
```

---

Done